js/util/actor.js

```diff
@@ -55,10 +61,16 @@ Actor.prototype.receive = function(message) {
     }
 };
 
-Actor.prototype.send = function(type, data, callback, buffers) {
+Actor.prototype.send = function(type, data, callback, buffers, targetMapId) {
     var id = callback ? this.mapId + ':' + this.callbackID++ : null;
     if (callback) this.callbacks[id] = callback;
-    this.postMessage({ mapId: this.mapId, type: type, id: String(id), data: data }, buffers);
+    this.postMessage({
+        targetMapId: targetMapId,
+        mapId: this.mapId,
```

Comment:
Why do we need to transfer both `mapId` and a `targetMapId`? Doesn't [this guard](https://github.com/mapbox/mapbox-gl-js/pull/3133/files#diff-07866509a19229697416fd9d0d2cae10R34) ensure they are equal?